google-account-plugin/src/com/google/cloud/tools/intellij/AccountPluginInitializationComponent.java

```diff
@@ -53,9 +56,26 @@ public void initComponent() {
     }
     if (!ApplicationManager.getApplication().isUnitTestMode()) {
       configureUsageTracking();
+      initUsageTracker();
     }
   }
 
+  @VisibleForTesting
+  void initUsageTracker() {
+    String platformPrefix = System.getProperty("idea.platform.prefix");
+    String trackingId = new PropertiesFilePluginFlags().getAnalyticsId();
+
+    Predicate<Object> userOptedIn = new Predicate<Object>() {
+      @Override
+      public boolean apply(@javax.annotation.Nullable Object object) {
+        return UsageTrackerManager.getInstance().hasUserOptedIn();
+      }
+    };
+
+    IdeaUsageTracker usageTracker = ServiceManager.getService(IdeaUsageTracker.class);
+    usageTracker.init(platformPrefix, trackingId, userOptedIn);
```

Comment:
That will do, actually. This revamp was started to make this class suitable for testing. So, using `Predicate` is drived by testing here, since `hasUserOptedIn()` requires accessing an actual IJ datastore (or something like that). The catch is to put this `hasUserOptedIn()` via Dependency Injection so that tests can conveniently make it return true or false. But perhaps, this DI may be too stretched, and there may be a way to mock the datastore in the background in testing???